### PR TITLE
Cleanup aws-parallelcluster-install::install and aws-parallelcluster-install::base

### PR DESCRIPTION
**Pull request by awslitvit**
### Description of changes
* [Cleanup aws-parallelcluster-install::install and aws-parallelcluster-install::base](https://github.com/aws/aws-parallelcluster-cookbook/commit/3aec67600afeb453fd9a8990becc75b2848bde60)
* [Validate aws-parallelcluster-install::base recipe](https://github.com/aws/aws-parallelcluster-cookbook/commit/72cfc5e1cac963e12585c3b5ca7e0f5f8e76f853)
* [Override parallelcluster-node-version for kitchen tests](https://github.com/aws/aws-parallelcluster-cookbook/commit/0da78c8e5c0ed8bd8ad092d20603d6c1fb52681d)
* [Add missing resources to kitchen.resources.yml](https://github.com/aws/aws-parallelcluster-cookbook/commit/65ed7f8bd9d4220cc2adbb6a6133a407ba3e3068)
* [Create docker_mock recipe to run before testing on Docker](https://github.com/aws/aws-parallelcluster-cookbook/commit/8651733a284314a91a4cd53ed89bddb91ec881f7)
* [Move python package versions from main::default.rb to common::common.rb](https://github.com/aws/aws-parallelcluster-cookbook/commit/5b8050d7b63c03683834366e24ae7dea4e6677af)
* [Avoid /tmp warning on redhat/ubi8 Docker](https://github.com/aws/aws-parallelcluster-cookbook/commit/4b9e067179d0e96bf99ed422ea03f074d6e6bb03)

### Tests
* Tested on Docker and on EC2

### References
About /tmp warning
* https://bugzilla.redhat.com/show_bug.cgi?id=2138434
* https://github.com/rubygems/rubygems/issues/4466#issuecomment-1301329937

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: kitchen.global.yml [hunk 1]
```diff
@@ -7,3 +7,4 @@ provisioner:
       region: us-east-1
       dcv:
         server: {}
+      parallelcluster-node-version: '3.4.1'
```
Comment (enrico-usai) [inline kitchen.global.yml:10]: should we add it to version-bump?

File: cookbooks/aws-parallelcluster-install/recipes/install.rb [hunk 2]
```diff
@@ -22,14 +22,29 @@
 
 node_attributes 'Generate export in json at /etc/chef/node_attributes.json'
 
+# == PLATFORM - BASE
 include_recipe 'aws-parallelcluster-install::base'
+
+# == PLATFORM - FEATURES
+include_recipe "aws-parallelcluster-install::nvidia" # NVIDIA and CUDA
+include_recipe "aws-parallelcluster-install::intel_mpi" unless virtualized?
+include_recipe "aws-parallelcluster-install::cloudwatch_agent"
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-install/recipes/install.rb:31]: this should be in the "observability" group.

File: cookbooks/aws-parallelcluster-install/recipes/install_mysql_client.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
   package_installer = value_for_platform(
     'default' => "yum install -y",
-    'ubuntu' => { 'default' => "apt install -y" }
+    'ubuntu' => { 'default' => "apt install -y --allow-downgrades" }
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-install/recipes/install_mysql_client.rb]: We're not sure we want to include this change, this is the reason we didnt' merge yet the patch: https://github.com/aws/aws-parallelcluster-cookbook/pull/1680

File: kitchen.global.yml (hunk 1 above)
Comment (awslitvit) [reply kitchen.global.yml:10]: No, not this way (take a look at the message on the commit). We will need to think more about it, this is temporary.

File: cookbooks/aws-parallelcluster-install/recipes/install.rb (hunk 2 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-install/recipes/install.rb:31]: Will fix later.

File: cookbooks/aws-parallelcluster-install/recipes/install_mysql_client.rb (hunk 3 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-install/recipes/install_mysql_client.rb]: Removing the commit.

**Review by enrico-usai — DISMISSED** [review]

**Review by enrico-usai — APPROVED** [review]

File: kitchen.docker.yml [hunk 4]
```diff
@@ -35,3 +35,5 @@ platforms:
   - name: redhat8
     driver:
       image: registry.access.redhat.com/ubi8/ubi
+      intermediate_instructions:
+        - RUN chmod +t /tmp
```
Comment (eantonin) [inline kitchen.docker.yml:39]: Thanks!!!!

File: cookbooks/aws-parallelcluster-common/attributes/common.rb [hunk 5]
```diff
@@ -36,3 +36,9 @@
 # cfn-bootstrap
 default['cluster']['cfn_bootstrap']['version'] = '2.0-10'
 default['cluster']['cfn_bootstrap']['package'] = "aws-cfn-bootstrap-py3-#{node['cluster']['cfn_bootstrap']['version']}.tar.gz"
+
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-common/attributes/common.rb:39]: This movement requires a change in the bump-version.sh

**Pull request merged**